compiler/hash-ast/src/ast.rs

```diff
@@ -149,21 +205,31 @@ pub enum RefKind {
     Normal,
 }
 
+/// A reference type.
+#[derive(Debug, PartialEq)]
+pub struct RefType<'c>(pub AstNode<'c, Type<'c>>);
+
+/// A raw reference type
+#[derive(Debug, PartialEq)]
+pub struct RawRefType<'c>(pub AstNode<'c, Type<'c>>);
+
+/// The existential type (`?`).
+#[derive(Debug, PartialEq)]
+pub struct ExistentialType;
+
+/// The type infer operator.
+#[derive(Debug, PartialEq)]
+pub struct InferType;
+
 /// A type.
 #[derive(Debug, PartialEq)]
 pub enum Type<'c> {
-    /// A concrete/"named" type.
     Named(NamedType<'c>),
-    /// A reference type.
-    Ref(AstNode<'c, Type<'c>>),
-    /// A raw reference type
-    RawRef(AstNode<'c, Type<'c>>),
-    /// A type variable.
+    Ref(RefType<'c>),
+    RawRef(RawRefType<'c>),
     TypeVar(TypeVar<'c>),
-    /// The existential type (`?`).
-    Existential,
-    /// The type infer operator.
-    Infer,
```

Comment:
Did you mean to remove comments on these variants?

Reply:
I did, because they all have a single member and the members have the same docs too. Otherwise docs would be duplicated for no real reason (in cargo-doc you can click on the member to view docs for it).